google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java

```diff
@@ -731,6 +740,65 @@ static String checkValidProperties(String uri) {
     return res;
   }
 
+  /**
+   * Executes a quick check to see if this connection can actually connect to a local emulator host
+   * or other (mock) test server, if the options point to localhost instead of Cloud Spanner.
+   */
+  private void checkLocalConnection() {
```

Comment:
Could we extract a class for this? I feel this class is very big already.

Reply:
Good point, I've moved it to a separate class.